server/src/main/java/org/opensearch/extensions/action/ExtensionActionUtil.java

```diff
@@ -0,0 +1,106 @@
+     * The Unicode UNIT SEPARATOR used to separate the Request class name and parameter bytes
+     */
+    public static final byte UNIT_SEPARATOR = (byte) '\u001F';
+
+    /**
+     * @param request r is an object of the "Remote Extension Action Request" class, containing information about the
+     * request being sent to the remote server. It is used to create a byte array containing the request data,
+     * which will be sent to the remote server.
+     * @return a byte array containing all the necessary information about the request to be sent to the remote server.
+     * This byte array is constructed using the class name of the request, a unit separator, and the request data itself.
+     */
+    public static byte[] createProxyRequestBytes(ExtensionActionRequest request) {
+        String requestClass = request.getClass().getName();
+        byte[] requestClassBytes = requestClass.getBytes(StandardCharsets.UTF_8);
+        return ByteBuffer.allocate(requestClassBytes.length + 1 + request.getRequestBytes().length)
+            .put(requestClassBytes)
+            .put(UNIT_SEPARATOR)
+            .put(request.getRequestBytes())
+            .array();
+    }
+
+    /**
+     * @param  requestBytes is a byte array containing the request data, used by the "createActionRequest"
+     * method to create an "ActionRequest" object, which represents the request model to be processed on the server.
+     * @return an "Action Request" object, which represents the request model for processing on the server,
+     * and is created using the request data stored in the byte array provided in the "requestBytes" parameter.
+     */
+    public static ActionRequest createActionRequest(ExtensionActionRequest requestBytes) {
+        int nullPos = delimPos(requestBytes.getRequestBytes());
+        String requestClassName = new String(Arrays.copyOfRange(requestBytes.getRequestBytes(), 0, nullPos + 1), StandardCharsets.UTF_8)
+            .stripTrailing();
+        ActionRequest actionRequest = null;
+        try {
+            Class<?> clazz = Class.forName(requestClassName);
+            Constructor<?> constructor = clazz.getConstructor(StreamInput.class);
+            StreamInput requestByteStream = StreamInput.wrap(
+                Arrays.copyOfRange(requestBytes.getRequestBytes(), nullPos + 1, requestBytes.getRequestBytes().length)
+            );
+            actionRequest = (ActionRequest) constructor.newInstance(requestByteStream);
+        } catch (Exception e) {
+            logger.debug("No request class [" + requestClassName + "] is available: " + e.getMessage());
+        }
+        return actionRequest;
+    }
+
+    /**
+     * @param actionRequest  is an instance of the Action Request class that is used to create an ExtensionActionRequest object.
+     * @return an Extension ActionRequest object based on an Action Request input.
+     */
+    public static ExtensionActionRequest createExtensionActionRequest(ActionRequest actionRequest) {
```

Comment:
you aren't doing anything with `actionRequest` here.  I'm not sure what this method is supposed to do.

See comment here: https://github.com/opensearch-project/opensearch-sdk-java/issues/607#issuecomment-1503811006

You want to replicate (at least some of) the code from the method `private static <T extends Writeable> byte[] convertParamsToBytes(T actionParams)` in https://github.com/opensearch-project/job-scheduler/pull/349.  Also look at the SDK HelloWorld RestRemoteHelloAction here: https://github.com/opensearch-project/opensearch-sdk-java/blob/main/src/main/java/org/opensearch/sdk/sample/helloworld/rest/RestRemoteHelloAction.java

Try to find the common parts of that code and the Job-Scheduler PR above and make that a method.